src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -1089,7 +1114,11 @@ private bool IsSubPathOfDeep(PackageItemRootViewModel path1, PackageItemRootView
                 {
                     return true;
                 }
-                else di2 = di2.Parent;
+                else
+                {
+                    if (di2.Parent.FullName.Length < di1.FullName.Length) return false;
```

Comment:
what is this condition, please add a comment - the original code appears to be from:
https://stackoverflow.com/questions/5617320/given-full-path-check-if-path-is-subdirectory-of-some-other-path-or-otherwise

so I would like to know what case you discovered?

Can you add a unit test for this case testing this function?

Reply:
Sure, if you look at the loop in which this method is called then you will understand it better. It is called inside the following loop:
```
foreach (var parent in items)
            {
                foreach(var child in items)
                {
                    if (IsSubPathOfDeep(parent.Value, child.Value))
                    {
                    ....
```
`items` consists all the folders in the package, so when we reach a point that child's(di2) parent's fullname becomes shorter than parent's(di1) fullname, at no point they will ever become equal after that since we are iterating on di2, which will only become shorter as the loop progresses.
I will try to add a test.

Reply:
can these paths ever be relative paths? Did this function work correctly with relative paths previously?

Reply:
From my testing this function always had absolute paths, I have added a test for this as well.